app/code/Magento/Catalog/Model/ResourceModel/Product/Gallery.php

```diff
@@ -513,4 +515,21 @@ public function countImageUses($image)
             );
         return count($this->getConnection()->fetchAll($select));
     }
+
+    /**
+     * Remove images position when product saved in default store
+     *
+     * @param int $entityId
+     * @param int $storeId
+     * @return $this
+     */
+    public function removeImagesPositionExcludeDefaultStore(int $entityId, int $storeId) : Gallery
```

Comment:
Hi @XxXgeoXxX can you please refactor the implementation to avoid adding public function to @api class?